src/ORM/Query.php

```diff
@@ -1181,22 +1203,51 @@ public function jsonSerialize()
     }
 
     /**
+     * Sets whether or not the ORM should automatically append fields.
+     *
+     * By default calling select() will disable auto-fields. You can re-enable
+     * auto-fields with this method.
+     *
+     * @param bool $value Set true to enable, false to disable.
+     * @return self
+     */
+    public function enableAutoFields($value)
+    {
+        $this->_autoFields = (bool)$value;
```

Comment:
Adding strict types onto documented types feels like the kind of thing we can fit within the scope of 4.x releases.

I can see how some people would interpret that as a BC break but I think we can argue that their code was breaking documented contracts before type checking was added.